### DISCUSSION TITLE
Link user by other attribute

### DISCUSSION THREAD

**Discussion by oculos** [keycloak/keycloak#28064] [Q&A] [answered]
My understand is that Keycloak identify that a user already exists based by e-mail (and, I suppose, by username).
Is it possible to do that by a custom attribute? That is, if a user from an IdP has the same attribute on the first login, then the account linking flow will take place?

**Comment by justin-tay** [comment, marked as answer]
I don't think this is possible out of the box. You will likely need to 

- Create a custom authenticator to replace the `IdpCreateUserIfUniqueAuthenticator`
- Configure `Authentication` to duplicate the `first browser login` flow to replace the `Create User If Unique` step with your custom authenticator
- Bind the `First browser login flow` with the flow you duplicated

https://github.com/keycloak/keycloak/blob/4bca804d5aa60308415ea5ef2727d3ddc3f7605a/services/src/main/java/org/keycloak/authentication/authenticators/broker/IdpCreateUserIfUniqueAuthenticator.java#L131-L147

**Reply by oculos** [reply]
Thanks a lot @justin-tay . I'll look into it.

**Comment by Cubox** [comment]
https://github.com/sd-f/keycloak-custom-attribute-idp-linking

This plugin might help
